tests/utils.py

```diff
@@ -839,9 +840,10 @@ def both_dirs_ready(n, scid):
         wait_for(lambda: both_dirs_ready(nodes[-1], scids[0]))
 
         # Make sure we have all node announcements, too (just check ends)
-        for n in nodes:
-            for end in (nodes[0], nodes[-1]):
-                wait_for(lambda: 'alias' in only_one(end.rpc.listnodes(n.info['id'])['nodes']))
+        if not announce_channels:
+            for n in nodes:
+                for end in (nodes[0], nodes[-1]):
+                    wait_for(lambda: 'alias' in only_one(end.rpc.listnodes(n.info['id'])['nodes']))
```

Comment:
Can't happen?  We have `if not wait_for_announce: return` above

Reply:
ack.